modules/Sources/Dependencies/AddressBookClient/AddressBookLiveKey.swift

```diff
@@ -199,34 +216,48 @@ extension AddressBookClient: DependencyKey {
             }
         }
 
+        var remoteStoreResult = RemoteStoreResult.notAttempted
         if storeAfterSync {
-            try storeContacts(syncedContacts, remoteStorage: remoteStorage, remoteStore: storeData)
+            remoteStoreResult = try storeContacts(syncedContacts, remoteStorage: remoteStorage, remoteStore: storeData)
         }
 
-        return syncedContacts
+        return (syncedContacts, remoteStoreResult)
     }
     
     private static func storeContacts(
         _ abContacts: AddressBookContacts,
         remoteStorage: RemoteStorageClient,
         remoteStore: Bool = true
-    ) throws {
+    ) throws -> RemoteStoreResult {
         // encrypt data
         let encryptedContacts = try AddressBookClient.encryptContacts(abContacts)
 
+        let filenameForEncryptedFile = try AddressBookClient.filenameForEncryptedFile()
+
         // store encrypted data to the local storage
         guard let documentsDirectory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first else {
             throw AddressBookClientError.documentsFolder
         }
-        
-        let filenameForEncryptedFile = try AddressBookClient.filenameForEncryptedFile()
 
         let fileURL = documentsDirectory.appendingPathComponent(filenameForEncryptedFile)
         try encryptedContacts.write(to: fileURL)
-        
+
         // store encrypted data to the remote storage
+        var isRemoteSuccess = remoteStore
```

Comment:
The `(false, true)` case in the switch isn't needed. I think it's clearer to do this (suggestion split over two comments to work around a GitHub limitation).
```suggestion
        var isRemoteSuccess = false
```
```suggestion
        // store encrypted data to the remote storage if requested
```

Reply:
Updated